### PR TITLE
refactor checksum loader task to optionally enqueue PrepareTransferJob

### PR DESCRIPTION
**Pull request by barmintor**

### CODE REVIEW

File: spec/atc/loaders/checksum_loader_spec.rb [hunk 1]
```diff
@@ -38,6 +46,15 @@
           expect(source_object.fixity_checksum_algorithm).to be_nil
         end
       end
+
+      context 'enqueue_successor is true' do
+        let(:enqueue_successor) { true }
+
+        it 'assigns the checksums' do
```
Comment (elohanlon) [inline spec/atc/loaders/checksum_loader_spec.rb:53]: It looks like this test is saying "when enqueue_successor is true it assigns the checksums", but it should assign the checksums whether or not enqueue_successor is true.  I know there's some additional verification going on in the `before` block, so this seems to be generally testing that nothing breaks when `enqueue_successor` is present, but would you mind rewording the `it` block to clarify what's being tested?

**Review by elohanlon — APPROVED** [review]
Approved!  Just had one suggestion about a test description.

File: spec/atc/loaders/checksum_loader_spec.rb (hunk 1 above)
Comment (barmintor) [reply spec/atc/loaders/checksum_loader_spec.rb:62]: in my mind that is part of what is being tested -  the perform_later expectation is in the before block. Do you have a recommendation?

Comment (elohanlon) [reply spec/atc/loaders/checksum_loader_spec.rb:62]: Maybe something like `it 'successfully assigns checksums and additionally calls PrepareTransferJob#perform_later'`, if that does a good enough job of describing your test intention?

File: spec/atc/loaders/checksum_loader_spec.rb [hunk 2]
```diff
@@ -38,6 +46,24 @@
...
+        it 'does not update the object' do
+          expect(source_object.fixity_checksum_value).to be_nil
+          expect(source_object.fixity_checksum_algorithm).to be_nil
+        end
+      end
+
+      context 'enqueue_successor is true' do
+        let(:enqueue_successor) { true }
+
+        it 'assigns the checksums' do
```
Comment (barmintor) [inline spec/atc/loaders/checksum_loader_spec.rb:62]:
```suggestion
        it 'successfully assigns checksums and additionally calls PrepareTransferJob#perform_later' do
```

**Pull request merged**